lib/grizzly/background_rssi_monitor.ex

```diff
@@ -0,0 +1,184 @@
+  def start_link(opts \\ []) do
+    classic_threshold = Keyword.get(opts, :classic_threshold, @default_classic_threshold)
+    lr_threshold = Keyword.get(opts, :lr_threshold, @default_lr_threshold)
+    sample_interval = Keyword.get(opts, :sample_interval, @default_sample_interval)
+    name = Keyword.get(opts, :name, __MODULE__)
+
+    GenServer.start_link(
+      __MODULE__,
+      [
+        classic_threshold: classic_threshold,
+        lr_threshold: lr_threshold,
+        sample_interval: sample_interval
+      ],
+      name: name
+    )
+  end
+
+  @impl GenServer
+  def init(opts) do
+    _ =
+      if opts[:sample_interval] != false do
+        Process.send_after(self(), :sample, opts[:sample_interval])
+      end
+
+    {:ok,
+     %{
+       classic_threshold: opts[:classic_threshold],
+       lr_threshold: opts[:lr_threshold],
+       sample_interval: opts[:sample_interval],
+       history: CircularBuffer.new(@history_size)
+     }}
+  end
+
+  @impl GenServer
+  def handle_call(:sample, _from, state) do
+    {status, state} = do_sample(state)
+    {:reply, status, state}
+  end
+
+  def handle_call(:get_averages, _from, state) do
+    [ch0, ch1, ch2, lr_primary, lr_secondary] = calculate_averages(state.history)
+
+    {:reply,
+     {:ok,
+      [
+        channels: [maybe_round(ch0), maybe_round(ch1), maybe_round(ch2)],
+        long_range_primary_channel: maybe_round(lr_primary),
+        long_range_secondary_channel: maybe_round(lr_secondary)
+      ]}, state}
+  end
+
+  @impl GenServer
+  def handle_info(:sample, state) do
+    {status, state} = do_sample(state)
+
+    if status == :ok do
+      Process.send_after(self(), :sample, state.sample_interval)
+    else
+      Process.send_after(self(), :sample, state.sample_interval * 3)
+    end
+
+    {:noreply, state}
+  end
+
+  defp do_sample(state) do
+    case get_sample() do
+      {:ok, sample} ->
+        state = %{state | history: CircularBuffer.insert(state.history, sample)}
+        set_or_clear_alarm(state)
+        {:ok, state}
+
+      {:error, :including} ->
+        {:error, state}
+
+      {:error, reason} ->
+        Logger.warning(
+          "[Grizzly.BackgroundRSSIMonitor] Failed to get background RSSI sample: #{inspect(reason)}"
+        )
+
+        {:error, state}
+    end
+  end
+
+  defp set_or_clear_alarm(state) do
+    [ch0, ch1, ch2, lr_primary, lr_secondary] = calculate_averages(state.history)
+
+    if (is_number(ch0) and ch0 >= state.classic_threshold) or
+         (is_number(ch1) and ch1 >= state.classic_threshold) or
+         (is_number(ch2) and ch2 >= state.classic_threshold) or
+         (is_number(lr_primary) and lr_primary >= state.lr_threshold) or
+         (is_number(lr_secondary) and lr_secondary >= state.lr_threshold) do
+      :alarm_handler.set_alarm({@alarm_id, []})
+    else
+      :alarm_handler.clear_alarm(@alarm_id)
+    end
+  end
+
+  defp calculate_averages(history) do
+    history
+    |> CircularBuffer.to_list()
```

Comment:
```suggestion
```
You shouldn't need this line since CircularBuffer implements the `Enumerable` protocol.